### PR TITLE
Update links to the proper releases page

### PR DESCRIPTION
**Pull request by marioevz**
Update links that point to the releases page from
`https://github.com/ethereum/staking-deposit-cli/releases`
to
`https://github.com/ethereum/eth2.0-deposit-cli/releases`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -532,4 +532,4 @@ python3 -m pytest .
 If you are trying to **build the binary** on macos with an M1 Mac and you are using pyenv to manage your python version. You'll probably need to reinstall a given python version using:
 ```
 env PYTHON_CONFIGURE_OPTS="--enable-framework" pyenv install 3.10.2
-```
\ No newline at end of file
+```
```
Comment (CarlBeek) [inline README.md:535]: for future reference, the change here is a newline at eof

**Comment by CarlBeek** [comment]
thanks @marioevz, i've changed the target to `dev` as that is our staging branch. it will then get included in the next release.

**Review by CarlBeek — APPROVED** [review]

**Comment by marioevz** [comment]
Hi @CarlBeek , just wanted to point out I made this change before I knew there was a rename in progress for the repo, is this still valid? The links were pointing to a 404 error page before but I am not sure is still the case with the rename.
Thanks!
Mario.

**Comment by CarlBeek** [comment]
No the changes were not relevant. I actually thought the changes were going the other way. (You replaced 'eth2.0' with 'staking', not the other way round.) That said I've already opened a PR that addresses it.

BTW, your 404 was probably because you were looking at `dev` not `master` which had unreleased changes.

**Pull request merged**